packages/ember/tests/routing/query_params_test/query_param_async_get_handler_test.js

```diff
@@ -0,0 +1,80 @@
+import {
+  Controller
+} from 'ember-runtime';
+
+import { QueryParamTestCase, moduleFor } from 'internal-test-helpers';
+
+moduleFor('Query Params - async get handler', class extends QueryParamTestCase {
+  get routerOptions() {
+    return {
+      location: 'test',
+      _getHandlerFunction() {
+        let getHandler = this._super(...arguments);
+        return (routeName) => Promise.resolve().then(() => getHandler(routeName));
```

Comment:
The problem with always deving in Chrome...will fix. I'll be sure to add a small setTimeout. Still planning to add several more tests.